### PR TITLE
Drop outdated `numpy_compat` functions

### PR DESCRIPTION
**Pull request by jakirkham**
These functions (`nancumsum` and `nancumprod`) have been removed as part of bumping the NumPy requirement to 1.13.0. So drop these fallback imports as these functions don't exist any more.

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
Hooray for dropping things!  +1 from me if tests pass.

File: dask/array/tests/test_array_core.py [hunk 1]
```diff
@@ -35,12 +35,7 @@
 from dask.array.utils import assert_eq, same_keys
 
 # temporary until numpy functions migrated
```
Comment (jrbourbeau) [inline dask/array/tests/test_array_core.py]: Can this comment also be dropped? Looks like it's referring to the `try/except`

```suggestion
```

File: dask/array/chunk.py [hunk 2]
```diff
@@ -67,11 +67,7 @@ def keepdims_wrapped_callable(x, axis=None, keepdims=None, *args, **kwargs):
 nansum = np.nansum
 nanprod = np.nanprod
 
-try:
-    from numpy import nancumprod, nancumsum
-except ImportError:  # pragma: no cover
-    nancumprod = npcompat.nancumprod
-    nancumsum = npcompat.nancumsum
+from numpy import nancumprod, nancumsum
```
Comment (jrbourbeau) [inline dask/array/chunk.py]: Neither `nancumprod` nor `nancumsum` appear to be used in this module. We may be able to drop this `import` entirely

File: dask/array/tests/test_array_core.py (hunk 1 above)
Comment (jakirkham) [reply dask/array/tests/test_array_core.py]: Yep, thanks for catching that @jrbourbeau. Have committed this change.

File: dask/array/chunk.py (hunk 2 above)
Comment (jrbourbeau) [reply dask/array/chunk.py]: Nvm, while `nancumprod` and `nancumsum` aren't used in this module they do get imported from this module in `dask/array/reductions.py`

Comment (jrbourbeau) [reply dask/array/chunk.py]: Doing

```python
nancumprod = np.nancumprod
nancumsum = np.nancumsum
```

should work and match the surrounding imports from numpy

Comment (jakirkham) [reply dask/array/chunk.py]: Makes sense. Have pushed such a commit. Thanks for the tip.

**Comment by jrbourbeau** [comment]
Will merge on green

**Comment by jrbourbeau** [comment]
Thanks @jakirkham

**Comment by jakirkham** [comment]
Thanks for all the timely feedback 😄

**Pull request merged**